File: DevEnvironment/Base/src/main/java/base/core/UnixPathBuilder.java
package base.core;

/**
 * The UnixPathBuilder class implements a builder of a unix path.
 */
public final class UnixPathBuilder extends AbstractPathBuilder {
    private static final String ROOT_DIRECTORY = UnixPath.ROOT_DIRECTORY;
    private static final String SEPARATOR = UnixPath.DIRECTORY_SEPARATOR;

    /**
     * The UnixPathBuilder constructor.
     */
    public UnixPathBuilder() {
        super(
            ROOT_DIRECTORY,
            SEPARATOR);
    }

    /**
     * The UnixPathBuilder constructor.
     */
    public UnixPathBuilder(String rootDirectory) {
        super(
            rootDirectory,
            SEPARATOR);
    }

    /**
     * The UnixPathBuilder constructor.
     */
    public UnixPathBuilder(String rootDirectory, String separator) {
        super(
            rootDirectory,
            separator);
    }
}

File: SystemCore/Base/core/UnixPathBuilder.cpp
#include "UnixPathBuilder.h"

#include "UnixPath.h"

using namespace base;

/**
 * Creates a new path builder.
 */
IPathBuilderSharedPtr UnixPathBuilder::Make()
{
    return std::make_shared<UnixPathBuilder>();
}

/**
 * The UnixPathBuilder constructor.
 */
UnixPathBuilder::UnixPathBuilder(const std::wstring& rootDirectory) :
    AbstractPathBuilder(
        rootDirectory,
        environment::unix::UnixPath::DirectorySeperator())
{
}

/**
 * The UnixPathBuilder destructor.
 */
UnixPathBuilder::~UnixPathBuilder()
{
}
